app/models/concerns/searchable.rb

```diff
@@ -0,0 +1,147 @@
+module Searchable
+  extend ActiveSupport::Concern
+
+  included do
+    include AlgoliaSearch
+
+    INDEX_NAME = [ENV["ALGOLIA_INDEX_PREFIX"], DOMAIN, name].compact.join("-").freeze
```

Comment:
I think having a constant defined here might be an issue.

It'll be set on the `Searchable` module even though it's in the `included` block, unless we use `base.const_set` (which might then still be an issue with referencing the constant in specs etc). How do you feel about moving the constant to be at the top of this file, so explicitly on the `Searchable` module, and then changing the references to `Vacancy::INDEX_NAME` throughout the app to `Searchable::INDEX_NAME`?